chain/chainservice.go

```diff
@@ -0,0 +1,40 @@
+package chain
+
+import (
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/btcutil/gcs"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/neutrino"
+	"github.com/lightninglabs/neutrino/banman"
+	"github.com/lightninglabs/neutrino/headerfs"
+)
+
+// NeutrinoChainService is an interface that encapsulates all the public
+// methods of a *neutrino.ChainService
+type NeutrinoChainService interface {
+	Start() error
+	GetBlock(chainhash.Hash, ...neutrino.QueryOption) (*btcutil.Block, error)
+	GetBlockHeight(*chainhash.Hash) (int32, error)
+	BestBlock() (*headerfs.BlockStamp, error)
+	GetBlockHash(int64) (*chainhash.Hash, error)
+	GetBlockHeader(*chainhash.Hash) (*wire.BlockHeader, error)
+	IsCurrent() bool
+	SendTransaction(*wire.MsgTx) error
+	GetCFilter(chainhash.Hash, wire.FilterType,
+		...neutrino.QueryOption) (*gcs.Filter, error)
+	GetUtxo(...neutrino.RescanOption) (*neutrino.SpendReport, error)
+	BanPeer(string, banman.Reason) error
```

Comment:
yes, so the first time I implemented this it was much slimmer, but I saw that this created a failure in LND.  This is because LND has some code that is directly accessing the `CS` field of a `NeutrinoClient`.  The consumer in LND expected the `CS` field to be the single `*neutrino.ChainService` concrete implementation, but in reality it was the slim interface.  So to be able to allow LND to update the version of `btcwallet` with this change I had to expose EVERY public method of `*neutrino.ChainService` event though a `NeutrinoClient` _only_ uses a few of the methods.

Reply:
Ohhhh, ok.  That's interesting.

I'd just note that lnd can assert that CS satisfies an even broader interface itself (the assertion applies to the underlying type, which can have more methods).  It seems like the lnd-btcwallet relationship is quite tight.  This partly explains why `CS` is even exported.